### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by redeemarr** [fastlib/fCWT#74]
Seems like FCWT::convolve lacks free(lastscalemem)

**Comment by fastlib** [comment]
Thanks for pointing it out! 

DWT is already been implemented very fast by multiple other libraries such as PyWavelets in Python and Wavelib!

**Comment by redeemarr** [comment]
> DWT is already been implemented very fast by multiple other libraries such as PyWavelets in Python and Wavelib!

Yes, but I need C++ one, so my only option is wavelib, but I'd prefer "fDWT" instead :)
Because fCWT interface is very simple and intuitive.
